Collect build directory paths while building each package into tmp/.package-builddir-info; used by automated custom binary packaging scripts.

diff --git a/include/package-dumpinfo.mk b/include/package-dumpinfo.mk
--- a/include/package-dumpinfo.mk
+++ b/include/package-dumpinfo.mk
@@ -5,6 +5,16 @@
 # See /LICENSE for more information.
 #
 
+PKG_BUILDDIR_INFO:=$(TMP_DIR)/.package-builddir-info
+
+define DumpBuildDirInfo
+	$(shell if [ -f $(PKG_BUILDDIR_INFO) ]; then \
+		sed -i 's|^$(PKG_NAME):.*|$(PKG_NAME): $(PKG_BUILD_DIR)|g' $(PKG_BUILDDIR_INFO); \
+		if [ -z "`grep -w ^$(PKG_NAME): $(PKG_BUILDDIR_INFO)`" ]; then \
+		echo "$(PKG_NAME): $(PKG_BUILD_DIR)" >> $(PKG_BUILDDIR_INFO); fi; \
+	else echo "$(PKG_NAME): $(PKG_BUILD_DIR)" >> $(PKG_BUILDDIR_INFO); fi)
+endef
+
 ifneq ($(DUMP),)
 
 dumpinfo: FORCE
diff --git a/include/package.mk b/include/package.mk
--- a/include/package.mk
+++ b/include/package.mk
@@ -224,6 +224,7 @@ define Build/DefaultTargets
   $(STAMP_PREPARED):
 	@-rm -rf $(PKG_BUILD_DIR)
 	@mkdir -p $(PKG_BUILD_DIR)
+	$(call DumpBuildDirInfo)
 	$(foreach hook,$(Hooks/Prepare/Pre),$(call $(hook))$(sep))
 	$(Build/Prepare)
 	$(foreach hook,$(Hooks/Prepare/Post),$(call $(hook))$(sep))
